docs/concepts/models.md

```diff
@@ -66,77 +74,96 @@ In this example, `User` is a model with two fields:
-The string `'123'` has been converted into an int as per the field type.
+1. `name` wasn't set when `user` was initialized, so the default value was used.
+   The [`model_fields_set`][pydantic.BaseModel.model_fields_set] attribute can be
+   inspected to check the field names explicitly set during instantiation.
 
-```py group="basic-model"
-assert user.name == 'Jane Doe'
-```
-
-`name` wasn't set when `user` was initialized, so it has the default value.
-
-```py group="basic-model"
-assert user.model_fields_set == {'id'}
-```
+Note that the string `'123'` was coerced to an integer and its value is `123`.
+More details on Pydantic's coercion logic can be found in the [Data Conversion](#data-conversion) section.
 
-The fields which were supplied when user was initialized.
+The model instance can be serialized using the [`model_dump`][pydantic.BaseModel.model_dump] method:
 
 ```py group="basic-model"
 assert user.model_dump() == {'id': 123, 'name': 'Jane Doe'}
 ```
 
-Either `.model_dump()` or `dict(user)` will provide a dict of fields, but `.model_dump()` can take numerous other
-arguments. (Note that `dict(user)` will not recursively convert nested models into dicts, but `.model_dump()` will.)
+Calling [dict][] on the instance will also provide a dictionary, but nested fields will not be
+recursively converted into dictionaries. [`model_dump`][pydantic.BaseModel.model_dump] also
+provides numerous arguments to customize the serialization result.
+
+By default, models are mutable and field values can be changed through attribute assignment:
 
 ```py group="basic-model"
 user.id = 321
 assert user.id == 321
 ```
 
-By default, models are mutable and field values can be changed through attribute assignment.
+!!! warning
+    When defining your models, watch out for naming collisions between your field name and its type annotation.
+
+    For example, the following will not behave as expected and would yield a validation error:
+
+    ```py test="skip"
+    from typing import Optional
+
+    from pydantic import BaseModel
+
+
+    class Boo(BaseModel):
+        int: Optional[int] = None
+
+
+    m = Boo(int=123)  # Will fail to validate.
+    ```
+
+    Because of how Python evaluates [annotated assignment statements][annassign], the statement is equivalent to `int: None = None`, thus
+    leading to a validation error.
 
 ### Model methods and properties
 
 The example above only shows the tip of the iceberg of what models can do.
 Models possess the following methods and attributes:
 
-* [`model_computed_fields`][pydantic.main.BaseModel.model_computed_fields]: a dictionary of the computed fields of this model instance.
-* [`model_construct()`][pydantic.main.BaseModel.model_construct]: a class method for creating models without running validation. See
+* [`model_validate()`][pydantic.main.BaseModel.model_validate]: Validates any object into a model. See [Validating data](#validating-data).
```

Comment:
Hmm, I feel like we want to be more specific here, maybe something more similar to the wording from the API docs, like "Validate the given object against the Pydantic model"

I'm tempted to add something about adhering to schema, but maybe that's too verbose for this brief explanation

Reply:
Sort of implied with "validation"